editor/store/selectors.js

```diff
@@ -88,7 +88,7 @@ export function isEditedPostNew( state ) {
  * @return {boolean} Whether unsaved values exist.
  */
 export function isEditedPostDirty( state ) {
-	return state.editor.isDirty;
+	return state.editor.isDirty || isSavingPost( state );
```

Comment:
Can you clarify why it should return true on mid-save?

Reply:
>Can you clarify why it should return true on mid-save?

The fix here is moving the change detection reset from `RESET_POST` to `UPDATE_POST`, so immediately upon hitting save, before the request has finished. This means that we can track changes that occur while the save is in-flight. Conversely, we need to make sure that if the user attempts to navigate away while the save is still pending, they're notified.

Another (better?) way to express this might be to consider whether there are any transactions in `redux-optimist` where `isDirty: true`; i.e. is there a possibility that if the transaction (save attempt) was to be reverted (save fail), would the user be put back into a state of dirtiness?

I'd also considered if we could track changes more like what we had, with it being a condition of the current state: This is simple enough to check if there are any entries in `state.editor.entries`, but `state.editor.blocksByUid` and `state.editor.blockOrder` is harder to track for changes (without comparing to an initial value).